content/blog/2018/01/2018-01-06-jep-200.adoc

```diff
@@ -0,0 +1,79 @@
+In a nutshell, this change is a security hardening measure
+to be less permissive about deserializing Java classes defined in the Java Platform or libraries bundled with Jenkins.
+For several years now, Jenkins has specifically _blacklisted_ certain classes and packages according to known or suspected exploits;
+now it will reject all classes not explicitly mentioned in a _whitelist_, or defined in Jenkins core or plugins.
+
+To the extent that advance testing of the impact of this change on popular plugins has been completed,
+most users (and even plugin developers) should not notice any difference.
+If you do encounter a `java.lang.SecurityException: Rejected: some.pkg.and.ClassName` in the Jenkins UI or logs,
+you may have found a case where an unusual plugin, or an unusual usage mode of a common plugin,
+violates the existing whitelist.
+This will be visible in the Jenkins system log as a message from `jenkins.security.ClassFilterImpl` like the following:
+
+----
+some.pkg.and.ClassName in file:/var/lib/jenkins/plugins/some-plugin-name/WEB-INF/lib/some-library-1.2.jar might be dangerous, so rejecting; see https://jenkins.io/redirect/class-filter/
+----
+
+where the link would direct you here.
+
+If you find such a case, please report it in the Jenkins issue tracker, under the appropriate plugin component.
+Link it to link:https://issues.jenkins-ci.org/browse/JENKINS-47736[JENKINS-47736] and add the `JEP-200` label.
+If at all possible, include complete steps to reproduce the problem from scratch.
+Jenkins developers will strive to evaluate the reason for the violation and offer a fix in the form of a core and/or plugin update.
+
+Assuming you see no particular reason to think that the class in question has dangerous deserialization semantics, which is rare,
+it is possible to work around the problem in your own installation as a temporary expedient.
+Simply make note of any class name(s) mentioned in such log messages,
+and run Jenkins with this startup option (details will depend on your installation method):
+
+----
+-Dhudson.remoting.ClassFilter=some.pkg.and.ClassName,some.pkg.and.OtherClassName
+----
+
+As a plugin developer encountering this kind of error,
+your first task is to ensure that it is reproducible in a functional (`JenkinsRule`) test
+when running
+
+[source,sh]
+----
+mvn test -Djenkins.version=2.105
+----
+
+// TODO ideally update to suggest actual merge version
+
+or whatever weekly or LTS version of Jenkins was reported to reproduce the error (or a newer version).
+You may need to run tests using an agent (e.g., `JenkinsRule.createSlave`) or force saves of plugin settings.
+
+Also make sure your `Jenkinsfile` specifies
+
+[source,groovy]
+----
+buildPlugin(jenkinsVersions: [null, '2.105'])
+----
+
+(again picking whatever version you need to test against)
+so that the test is included during CI builds, even while your minimum core baseline predates JEP-200.
+
+You then have several choices for fixing the problem:
```

Comment:
I would explicitly split workarounds to two sections:

* Workarounds for Jenkins admins
* Workarounds for developers

Reply:
agreed

Reply:
(resolved)